python/sparkdl/param/converters.py

```diff
@@ -0,0 +1,176 @@
+        """
+        Convert a value to a :py:obj:`tf.Graph` object, if possible.
+        """
+        if not isinstance(value, tf.Graph):
+            raise TypeError("Could not convert %s to tf.Graph" % type(value))
+        return value
+
+    @staticmethod
+    def asColumnToTensorNameMap(value):
+        """
+        Convert a value to a column name to :py:obj:`tf.Tensor` name mapping
+        as a sorted list of string pairs, if possible.
+        """
+        if not isinstance(value, dict):
+            err_msg = "Could not convert [type {}] {} to column name to tf.Tensor name mapping"
+            raise TypeError(err_msg.format(type(value), value))
+
+        # Conversion logic after quick type check
+        strs_pair_seq = []
+        for _maybe_col_name, _maybe_tnsr_name in value.items():
+            # Check if the non-tensor value is of string type
+            _check_is_str(_maybe_col_name)
+            # Check if the tensor name looks like a tensor name
+            _check_is_tensor_name(_maybe_tnsr_name)
+            strs_pair_seq.append((_maybe_col_name, _maybe_tnsr_name))
+
+        return sorted(strs_pair_seq)
+
+    @staticmethod
+    def asTensorNameToColumnMap(value):
+        """
+        Convert a value to a :py:obj:`tf.Tensor` name to column name mapping
+        as a sorted list of string pairs, if possible.
+        """
+        if not isinstance(value, dict):
+            err_msg = "Could not convert [type {}] {} to tf.Tensor name to column name mapping"
+            raise TypeError(err_msg.format(type(value), value))
+
+        # Conversion logic after quick type check
+        strs_pair_seq = []
+        for _maybe_tnsr_name, _maybe_col_name in value.items():
+            # Check if the non-tensor value is of string type
+            _check_is_str(_maybe_col_name)
+            # Check if the tensor name looks like a tensor name
+            _check_is_tensor_name(_maybe_tnsr_name)
+            strs_pair_seq.append((_maybe_tnsr_name, _maybe_col_name))
+
+        return sorted(strs_pair_seq)
+
+    @staticmethod
+    def toTFHParams(value):
+        """ Convert a value to a :py:obj:`tf.contrib.training.HParams` object, if possible. """
+        if not isinstance(value, tf.contrib.training.HParams):
+            raise TypeError("Could not convert %s to TensorFlow HParams" % type(value))
+
+        return value
+
+    @staticmethod
+    def toTFTensorName(value):
+        """ Convert a value to a :py:obj:`tf.Tensor` name, if possible. """
+        if isinstance(value, tf.Tensor):
+            return value.name
+        try:
+            _maybe_tnsr_name = TypeConverters.toString(value)
+            _check_is_tensor_name(_maybe_tnsr_name)
+            return _maybe_tnsr_name
+        except Exception as exc:
+            err_msg = "Could not convert [type {}] {} to tf.Tensor name. {}"
+            raise TypeError(err_msg.format(type(value), value, exc))
+
+    @staticmethod
+    def buildCheckList(supportedList):
```

Comment:
not sure what the new name is conveying. why the change?

Reply:
let's try not to change existing code that is not relevant to the current PR unless it's really trivial (not just implementation but also the reason). it adds to the review & discussion time unnecessarily.

Reply:
Sorry it's really hard to see previous comments in github. But regardless, `buildCheckList` doesn't invoke any sense to me that it's building one of these Converters that checks against a list. maybe `buildSupportedItemConverter` or `buildSupportedItemChecker`. Still would go with Converter because it's the common terminology in mllib for where these are used (as the last parameter "TypeConverter" in Param definitions).